src/catalog/src/durable/persist.rs

```diff
@@ -1431,18 +1431,19 @@ impl DurableCatalogState for PersistCatalogState {
         Ok(())
     }
 
-    // TODO(jkosh44) For most modifications we delegate to transactions to avoid duplicate code.
-    // This is slightly inefficient because we have to clone an entire snapshot when we usually
-    // only need one part of the snapshot. A Potential mitigation against these performance hits is
-    // to utilize `CoW`s in `Transaction`s to avoid cloning unnecessary state.
-
     #[mz_ore::instrument]
     async fn prune_storage_usage(
         &mut self,
         retention_period: Option<Duration>,
         boot_ts: mz_repr::Timestamp,
-    ) -> Result<(), CatalogError> {
+    ) -> Result<Vec<memory::objects::StateUpdate>, CatalogError> {
         self.sync_to_current_upper().await?;
+
+        if self.is_read_only() {
+            self.confirm_leadership().await?;
```

Comment:
Why is this necessary? Doesn't sync_to_current_upper guarantee this? It's documented as

> "Returns an error if this instance has been fenced out."

Update: ok I see it's here because it's just getting moved from where it already is. However, from a correctness perspective it's possible it was never needed?

Reply:
> However, from a correctness perspective it's possible it was never needed?

Yes, good catch. This seems totally redundant, I'll remove it.

Reply:
Done